src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,171 @@
+import { getLoadingStatus } from '../data/selectors';
+import { enableDisableApp } from '../data/thunks';
+import AppConfigFormDivider from '../discussions/app-config-form/apps/shared/AppConfigFormDivider';
+import { PagesAndResourcesContext } from '../PagesAndResourcesProvider';
+import messages from './messages';
+
+function AppSettingsForm({ formikProps, children }) {
+  return children && (
+    <TransitionReplace>
+      {formikProps.values.enabled
+        ? (
+          <React.Fragment key="app-enabled">
+            {children(formikProps)}
+          </React.Fragment>
+        ) : (
+          <React.Fragment key="app-disabled" />
+        )}
+    </TransitionReplace>
+  );
+}
+
+AppSettingsForm.propTypes = {
+  formikProps: PropTypes.shape({
+    values: PropTypes.shape({ enabled: PropTypes.bool.isRequired }),
+  }).isRequired,
+  children: PropTypes.func,
+};
+
+AppSettingsForm.defaultProps = {
+  children: null,
+};
+
+function AppSettingsModal({
+  intl,
+  appId,
+  title,
+  children,
+  initialValues,
+  validationSchema,
+  onClose,
+  onSettingsSave,
+  enableAppLabel,
+  enableAppHelp,
+  learnMoreURL,
+  learnMoreText,
+}) {
+  const { courseId } = useContext(PagesAndResourcesContext);
+  const loadingStatus = useSelector(getLoadingStatus);
+  const appInfo = useModel('courseApps', appId);
+  const dispatch = useDispatch();
+  const handleFormSubmit = (values) => {
+    // If the app's enabled/disabled loadingStatus has changed, set that first.
+    if (appInfo.enabled !== values.enabled) {
+      dispatch(enableDisableApp(courseId, appInfo.id, values.enabled, true));
+    }
+    // Call the submit handler for the settings component to save its settings
+    if (onSettingsSave) {
+      onSettingsSave();
+    }
+  };
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL} target="_blank">{learnMoreText}</Hyperlink>
+  );
+
+  return (
+    <ModalLayer
+      isOpen
+      closeText={intl.formatMessage(messages.cancel)}
+      dialogClassName="modal-dialog-centered modal-lg"
+      onClose={onClose}
+    >
+      <div
+        role="dialog"
+        aria-label={title}
+        className="bg-white d-flex flex-column mw-xs p-3"
+      >
+        {
+          loadingStatus === RequestStatus.SUCCESSFUL && (
+            <Formik
+              initialValues={{
+                enabled: !!appInfo?.enabled,
+                ...initialValues,
+              }}
+              validationSchema={
+                Yup.object()
+                  .shape({
+                    enabled: Yup.boolean(),
+                    ...validationSchema,
+                  })
+              }
+              onSubmit={handleFormSubmit}
+            >{(formikProps) => (
+              <Form onSubmit={formikProps.handleSubmit}>
+                <h3>{title}</h3>
+                <AppConfigFormDivider />
+                <FormSwitchGroup
+                  id={`enable-${appId}-toggle`}
```

Comment:
Just checking why this ID is needed? The general perception for adding`id`s is to things if we need to use the ID somewhere e.g. tests or referencing. I can't see any usage of this.

Reply:
It  is required by the FormSwitchGroup so I added it. I will update the component to make it optional.

Reply:
I see, thank you.

Reply:
Actually, reading through the comments in FormSwitchGroup it seems that currently you need to attach a label to the switch component.